### ISSUE TITLE
Refactor Code in Python or Golang

### ISSUE THREAD

**Issue by BethanyG** [codebuddiesdotorg/greetbot#92]
Refactor the code in Python or Golang for ease of maintenance.

**Comment by stain88** [comment]
Obligatory comment about ruby

**Comment by BethanyG** [comment]
If Ruby's your passion @stain88 - I won't get in your way.  I'd be sad to give up the project entirely (I don't know any Ruby, and I don't currently have the bandwidth to learn) - but if you are on fire about it, I can re-pull the code into my repo and refactor it as and individual project....then you can go ahead a re-write it as Ruby for the community.

**Comment by stain88** [comment]
nah, not to the extent of forcing you to drop the project. Playing around with a ruby bot on my own to see how that works.
I'll start having a look at python, since bill is going the go route

**Comment by angelocordon** [comment]
I’m going to keep it on Node just to throw a monkey wrench in this whole
thread :D

On Jul 31, 2018 at 1:13 PM, Marc Baghdadi <notifications@github.com> wrote:


nah, not to the extent of forcing you to drop the project. Playing around
with a ruby bot on my own to see how that works.
I'll start having a look at python, since bill is going the go route

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/codebuddies/greetbot/issues/92#issuecomment-409352494>,
or mute the thread

**Comment by BethanyG** [comment]
@angelocordon - the thought would be that we'd keep a version on Node as long as there were slash commands being added, or there was active development happening by one or more participants.  Meanwhile, we'd be looking at re-writing the codebase on a different branch/in a different repo.  It is small enough at this point that it (ought) to be straightforward.  

 @stain88 - totally up to you (you could still do it in Ruby...).  If you'd like to go the Python route, there is Flask or Django for the backend.  We could use Redis for storing the message content - or even something like postgres.  Django comes with SQLLite, as well.  There are also numerous bot frameworks...but I am assuming that we'd do it from "scratch" or "scratch-ish" - although there might be something super-fun out there (it's been a while since I looked into it).  Another thought/pattern would be to AWS-Lambda-ize the way @bill is doing with buddybot.   Many options....
